dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -209,7 +213,29 @@ namespace dwa_local_planner {
     delete dsrv_;
   }
 
+  void DWAPlannerROS::resetBestEffort() {
+    latched_inner_goal_ = false;
+    prev_vel_dir_ = 0;
+    oscillating_ = false;
+  }
+
+  bool DWAPlannerROS::finishedBestEffort() {
+    std::vector<geometry_msgs::PoseStamped> plan;
+    if (!planner_util_.getLocalPlan(current_pose_, plan)) {
```

Comment:
Q/S: this is rather expensive;  makes sense to store as an attribute? (a bit ugly, I know)

Reply:
Fixed